tesseract-server/src/handlers/logic_layer/metadata.rs

```diff
@@ -0,0 +1,198 @@
+use actix_web::{
+    AsyncResponder,
+    FutureResponse,
+    HttpRequest,
+    HttpResponse,
+    Path,
+};
+use failure::Error;
+use futures::future::{self, Future};
+use lazy_static::lazy_static;
+use log::*;
+use serde_derive::{Serialize, Deserialize};
+use serde_qs as qs;
+
+use crate::app::AppState;
+use crate::logic_layer::{LogicLayerConfig};
+use crate::util::boxed_error;
+
+use tesseract_core::format::{format_records, FormatType};
+use tesseract_core::names::LevelName;
+
+use super::super::util;
+
+
+/// Handles default members query when a format is not specified.
+/// Default format is CSV.
+pub fn logic_layer_members_default_handler(
+    (req, _cube): (HttpRequest<AppState>, Path<()>)
+) -> FutureResponse<HttpResponse>
+{
+    get_members(req, "jsonrecords".to_owned())
+}
+
+
+/// Handles members query when a format is specified.
+pub fn logic_layer_members_handler(
+    (req, cube_format): (HttpRequest<AppState>, Path<(String)>)
+) -> FutureResponse<HttpResponse>
+{
+    get_members(req, cube_format.to_owned())
+}
+
+
+/// Performs members query.
+pub fn get_members(
+    req: HttpRequest<AppState>,
+    format: String,
+) -> FutureResponse<HttpResponse>
+{
+    let format = format.parse::<FormatType>();
+    let format = match format {
+        Ok(f) => f,
+        Err(err) => return boxed_error(err.to_string()),
+    };
+
+    info!("Format: {:?}", format);
+
+    let query = req.query_string();
+    let schema = req.state().schema.read().unwrap();
+    let debug = req.state().debug;
+
+    let logic_layer_config: Option<LogicLayerConfig> = match &req.state().logic_layer_config {
+        Some(llc) => Some(llc.read().unwrap().clone()),
+        None => None
+    };
+
+    lazy_static!{
+        static ref QS_NON_STRICT: qs::Config = qs::Config::new(5, false);
+    }
+
+    let members_query = match QS_NON_STRICT.deserialize_str::<MembersQueryOpt>(query) {
+        Ok(q) => q,
+        Err(err) => return boxed_error(err.to_string())
+    };
+
+    let mut cube_name = members_query.cube.clone();
+    let mut level_name: Option<LevelName> = None;
+
+    match &logic_layer_config {
```

Comment:
We should probably investigate some ways to reduce `Option` nesting hell. In the meantime, can we use `if let Some` so that at least we don't have the `None` side of the match?